src/sorting/shell_sort.rs

```diff
@@ -0,0 +1,52 @@
+pub fn shell_sort<T: Ord + Copy>(values: &mut Vec<T>) {
+    // shell sort works by swiping the value at a given gap and decreasing the gap to 1
+    fn insertion<T: Ord + Copy>(values: &mut Vec<T>, start: usize, gap: usize) {
+        for i in ((start + gap)..values.len()).step_by(gap) {
+            let val_current = values[i];
+            let mut pos = i;
+            // make swaps
+            while pos >= gap && values[pos - gap] > val_current {
+                values[pos] = values[pos - gap];
```

Comment:
Maybe you might want to take a look [`slice.swap()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap)?